pkg/azurefile/controllerserver.go

```diff
@@ -853,32 +853,67 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 		}, nil
 	}
 
-	shareURL, err := d.getShareURL(ctx, sourceVolumeID, req.GetSecrets())
+	// if len(req.GetSecrets()) > 0 {
+	// 	shareURL, err := d.getShareURL(ctx, sourceVolumeID, req.GetSecrets())
+	// 	if err != nil {
+	// 		return nil, status.Errorf(codes.Internal, "failed to get share url with (%s): %v", sourceVolumeID, err)
+	// 	}
+
+	// 	snapshotShare, err := shareURL.CreateSnapshot(ctx, azfile.Metadata{snapshotNameKey: snapshotName})
+	// 	if err != nil {
+	// 		return nil, status.Errorf(codes.Internal, "create snapshot from(%s) failed with %v, shareURL: %q", sourceVolumeID, err, shareURL)
+	// 	}
+
+	// 	klog.V(2).Infof("Created share snapshot: %s", snapshotShare.Snapshot())
+
+	// 	properties, err := shareURL.GetProperties(ctx)
+	// 	if err != nil {
+	// 		return nil, status.Errorf(codes.Internal, "failed to get snapshot properties from (%s): %v", snapshotShare.Snapshot(), err)
+	// 	}
+
+	// 	tp := timestamppb.New(properties.LastModified())
+	// 	if tp == nil {
+	// 		return nil, status.Errorf(codes.Internal, "Failed to convert timestamp(%v)", properties.LastModified())
+	// 	}
+
+	// 	createResp := &csi.CreateSnapshotResponse{
+	// 		Snapshot: &csi.Snapshot{
+	// 			SizeBytes:      volumehelper.GiBToBytes(int64(properties.Quota())),
+	// 			SnapshotId:     sourceVolumeID + "#" + snapshotShare.Snapshot(),
+	// 			SourceVolumeId: sourceVolumeID,
+	// 			CreationTime:   tp,
+	// 			// Since the snapshot of azurefile has no field of ReadyToUse, here ReadyToUse is always set to true.
+	// 			ReadyToUse: true,
+	// 		},
+	// 	}
+
+	// 	isOperationSucceeded = true
+	// 	return createResp, nil
+	// } else {
+	snapshotShare, err := d.cloud.FileClient.WithSubscriptionID(subsID).CreateFileShare(ctx, rgName, accountName, &fileclient.ShareOptions{Name: fileShareName, RequestGiB: defaultAzureFileQuota, Metadata: map[string]*string{snapshotNameKey: &snapshotName}}, "snapshots")
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to get share url with (%s): %v", sourceVolumeID, err)
+		return nil, status.Errorf(codes.Internal, "create snapshot from(%s) failed with %v, accountName: %q", sourceVolumeID, err, accountName)
 	}
 
-	snapshotShare, err := shareURL.CreateSnapshot(ctx, azfile.Metadata{snapshotNameKey: snapshotName})
+	klog.V(2).Infof("Created share snapshot: %s", snapshotShare.Header.Get("x-ms-snapshot"))
```

Comment:
snapshotShare.Header.Get("x-ms-snapshot") it's always empty, use snapshotShare.SnapshotTime.Time instead.